session.go

```diff
@@ -847,6 +853,14 @@ func (s *Session) Login(cred *Credential) error {
 	defer socket.Release()
 
 	credCopy := *cred
+	if cred.Certificate != nil && cred.Username != "" {
+		return fmt.Errorf("Failed to login, both certifcate and credentials are given.")
```

Comment:
It's more efficient to use `errors.New()` when you don't require a formatter, and errors typically (should) start with a lowercase letter.

Have a quick read of https://github.com/golang/go/wiki/CodeReviewComments though the existing mgo code doesn't exactly follow it